File: js/src/model/matching/Match.js
/**
 * Match result presentation
 * @author tanzhenlin
 * @date 2022/6/29 11:22
 **/
export default class Match {
    rule;
    items;

    constructor (rule, items) {
        this.rule = rule;
        this.items = items;
    }

    totalDiscount() {
        if (!this.items || this.items.length === 0 || !this.rule) return 0;
        return this.rule.discount(this.items);
    }

    count() {
        if (!this.items) return 0;
        return this.items.length;
    }

    totalPrice() {
        if (!this.items) return 0;
        return this.items.map(t => t.price).reduce((a, b) => a + b, 0);
    }
}

File: java/src/main/java/com/github/thinhunan/wonder8/promotion/rule/model/strategy/Match.java
/**
 * Match result presentation
 * @author tanzhenlin
 * @date 2022/6/29 11:22
 **/
package com.github.thinhunan.wonder8.promotion.rule.model.strategy;

import com.github.thinhunan.wonder8.promotion.rule.model.Item;
import com.github.thinhunan.wonder8.promotion.rule.model.Rule;

import java.util.List;

public class Match {
    Rule rule;
    List<Item> items;

    public Rule getRule() {
        return rule;
    }

    public void setRule(Rule rule) {
        this.rule = rule;
    }

    public List<Item> getItems() {
        return items;
    }

    public void setItems(List<Item> items) {
        this.items = items;
    }

    public Match(Rule rule, List<Item> items) {
        this.rule = rule;
        this.items = items;
    }

    public int totalDiscount(){
        if(items == null || items.size() == 0 || rule == null) return 0;
        return rule.discount(items);
    }

    public int count(){
        if(items == null) return 0;
        return items.size();
    }

    public int totalPrice(){
        if(items == null) return 0;
        return items.stream().map(Item::getPrice).reduce(0,Integer::sum);
    }
}
